osu.Game/Screens/Select/BeatmapCarousel.cs

```diff
@@ -301,8 +303,13 @@ private void removeGroup(BeatmapGroup group)
             foreach (var p in group.BeatmapPanels)
                 panels.Remove(p);
 
-            scrollableContent.Remove(group.Header);
-            scrollableContent.RemoveRange(group.BeatmapPanels);
+            // Must check if panels aren't already removed because they're offscreen

```

Comment:
Why? Doesn't `Remove` simply return false when attempting to remove something that's not currently in the container? Makes for cleaner code imho.